### ISSUE TITLE
Node child forks can't use the debugger

### ISSUE THREAD

**Issue by romainbessugesmeusy** [josdejong/workerpool#19]
Hi Jos,

It looks like I can't use node debugger while spawning child processes, because each worker try to use the same port. Example:

```
Debugger listening on 127.0.0.1:51884
Error: listen EADDRINUSE 127.0.0.1:51884
    at Object.exports._errnoException (util.js:1023:11)
    at exports._exceptionWithHostPort (util.js:1046:20)
    at Agent.Server._listen2 (net.js:1261:14)
    at listen (net.js:1297:10)
    at doListening (net.js:1396:7)
    at _combinedTickCallback (internal/process/next_tick.js:77:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

It seems that child forks have to be created with the `--debugger={port}`. My first guess would be to change this line (WorkerHandler.js:79):

```javascript
    this.worker = node.require('child_process').fork(this.script);
```

What do you think?

Thanks again for your work, cheers from France

**Comment by romainbessugesmeusy** [comment]
I think it's interesting to mention this discussion on the IntelliJ / WebStorm board

https://intellij-support.jetbrains.com/hc/en-us/community/posts/206193339-Debugging-node-child-processes

**Comment by josdejong** [comment]
Thanks for reporting Romain!

> It seems that child forks have to be created with the --debugger={port}

Wouldn't that always run the childs in debug mode, also if you just want to run in production?

**Comment by romainbessugesmeusy** [comment]
Yeah I fear you're right. Furthermore I haven't found yet how to attach a remote debugger to child process inside Webstorm... Maybe this could be passed when creating the pool, as a secondary argument.

```javascript
var workerpool = require('workerpool');

// create a worker pool using an external worker script
var pool = workerpool.pool(__dirname + '/myWorker.js', {
   node: {
       childProcessExecArgs: (childIndex) => return '--debugger=' + (53027 + childIndex)
   }
});
 ```

**Comment by josdejong** [comment]
hm that's odd. I can hardly believe we're the first stumbling upon this problem.

Maybe we could check whether the `--debugger` option is in the arguments of the main process, and only then add it in the `childProcessExecArgs` with a unique port. Or is that what you mean?

From the [thead you mentioned](https://intellij-support.jetbrains.com/hc/en-us/community/posts/206193339-Debugging-node-child-processes?page=1#community_comment_206895729) I understand that it should be enough to pass an argument `--debug-brk` which will let node.js automatically pick a free port.

> Oh, it could be simplified in the following way (forgot to mention it in the first place):
> ```
> var childProc= fork('./public/LG/lg.js',{execArgv: ['--debug-brk']})
> ```
> In that case node automatically finds a free port.

@romainbessugesmeusy would you be interested in doing some experimenting with this to find a working solution for workerpool?

**Comment by romainbessugesmeusy** [comment]
Yeah I'll look into that 👌

**Comment by josdejong** [comment]
:+1:

**Comment by josdejong** [comment]
@romainbessugesmeusy did you already found some time to look into the debugging issue?

**Comment by josdejong** [comment]
This issue should have been fixed now in `v2.2.1` (see #21)
